Add mcxn236 i3c support

NeilChen93 opened the pull request:
Add mcxn236 i3c support

NeilChen93 committed:
dts: arm/nxp: Add i3c nodes to NXP MCXN23x dtsi file

Add i3c nodes to NXP MCXN23x dtsi file

Signed-off-by: Neil Chen <cheng.chen_1@nxp.com>

danieldegrasse commented on boards/nxp/frdm_mcxn236/frdm_mcxn236.dts:
```diff
@@ -131,6 +131,10 @@
 	status = "okay";
 };
 
+&i3c1 {
```
Add I3C to the board YAML as well, like here: https://github.com/zephyrproject-rtos/zephyr/blob/e1389fa2cff66c4f0c1ac5d0b1b932ef0e3faf06/boards/nxp/frdm_mcxn947/frdm_mcxn947_mcxn947_cpu0.yaml#L31

Also, does this board have any sensor on the I3C bus we could enable here, to provide an easy way to validate the bus works?

NeilChen93 replied:
Hi @danieldegrasse , frdmmcxn236 board don't have I3C sensor.

DerekSnell replied:
FYI, some Freedom boards include the P3T1755 I3C temperature sensor.  Mahesh just enabled this sensor in [Add I3C support for FRDM-MCXN947](https://github.com/zephyrproject-rtos/zephyr/pull/79738).  The N947 is a very similar SOC and shares the same I3C peripheral, so the driver is validated on that board.

For other boards without this sensor, NXP offers the [P3T1755DP-ARD Arduino shield](https://www.nxp.com/design/design-center/development-boards-and-designs/analog-toolbox/arduino-shields-solutions/p3t1755dp-arduino-shield-evaluation-board:P3T1755DP-ARD).  @hakehuang has one of these shields so we can test boards without I3C sensors if needed.

Best regards

hakehuang replied:
in P3T1755, the pin on arduino is lay as below: is it possible to align? as I can see the P1_11 is not compatible
```
![image](https://github.com/user-attachments/assets/08bd8141-8367-4f10-8d50-a054cfea3f87)

```

dleach02 commented:
Update the board doc page with this added feature

NeilChen93 commented:
Hi @hakehuang , I already enabled MCXN236 I3C support according to Lucien's PR. But I also don't have local environment to test, could you help test it on your side? Thanks a lot.

hakehuang commented on samples/sensor/thermometer/sample.yaml:
```diff
@@ -15,6 +15,8 @@ tests:
       - platform:mimxrt1180_evk/mimxrt1189/cm33:SHIELD=p3t1755dp_ard_i3c
       - platform:mimxrt1180_evk/mimxrt1189/cm7:SHIELD=p3t1755dp_ard_i3c
+      - platform:frdm_mcxn236:SHIELD=p3t1755dp_ard_i2c
+      - platform:frdm_mcxn236:SHIELD=p3t1755dp_ard_i3c
```
sorry, I miss check luient's pr for this part, you can't assign the same platform with i2c, and i3c at the same time, this will cause problem, you need add another test node like below
```
common:
    filter: dt_alias_exists("ambient-temp0")
    harness: console
    harness_config:
      fixture: sensor_ambient_temp
      type: multi_line
      regex:
        - "Thermometer Example (.*)"
        - "Temperature device is 0x[0-9|a-z]+, name is [a-z|0-9]+@[a-z|0-9]+"
        - "Temperature is [0-9|.]+°C"
tests:
  sample.sensor.thermometer:
    tags: sensors
    fixture:
      - fixture_i2c_sensor 
    integration_platforms:
      - nrf52840dk/nrf52840   # mcp9700a
      - frdm_k22f             # tcn75a
      - robokit1              # ntc_thermistor
      - adi_eval_adin1110ebz  # adt7420
    extra_args:
      - platform:mimxrt1180_evk/mimxrt1189/cm33:SHIELD=p3t1755dp_ard_i2c
      - platform:mimxrt1180_evk/mimxrt1189/cm7:SHIELD=p3t1755dp_ard_i2c
      - platform:frdm_mcxn236:SHIELD=p3t1755dp_ard_i2c
  sample.sensor.thermometer_i3c:
    tags: sensors
    integration_platforms:
      - frdm_mcxn947/mcxn947/cpu0 # p3t1755
    depends_on:
    - i3c
    - sensor
    filter: dt_alias_exists("ambient-temp0")
    fixture: fixture_i3c_sensor 
    extra_args:
      - platform:mimxrt1180_evk/mimxrt1189/cm33:SHIELD=p3t1755dp_ard_i3c
      - platform:mimxrt1180_evk/mimxrt1189/cm7:SHIELD=p3t1755dp_ard_i3c
      - platform:frdm_mcxn236:SHIELD=p3t1755dp_ard_i3c
```

hakehuang replied:
also please add i3c support to `boards/nxp/frdm_mcxn236/frdm_mcxn236.yaml`

hakehuang commented:
@NeilChen93 , the i2c node works fine as below,
```
Temperature is 24.2¡ãC
Temperature is 24.2¡ãC
Temperature is 24.2¡ãC
Temperature is 24.2¡ãC
Temperature is 24.2¡ãC
Temperature is 24.1¡ãC
Temperature is 24.2¡ãC
Temperature is 24.2¡ãC
Temperature is 24.2¡ãC
Temperature is 24.2¡ãC
Temperature is 24.2¡ãC
Temperature is 24.1¡ãC
```
but I3C does not, once enable i2c mode the uart console output mess chars, please check.

```
▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒
```

NeilChen93 committed:
boards: nxp: frdm_mcxn236: Support i3c for NXP frdm_mcxn236 board

Support i3c for NXP frdm_mcxn236 board.

Signed-off-by: Neil Chen <cheng.chen_1@nxp.com>

NeilChen93 committed:
samples: sensor: Add support for temperature sensor on FRDM_MCXN236

Add new test sample.sensor.thermometer_i3c, can't assign the same
platform with i2c, and i3c at the same time.
Support the shield on FRDM_MCXN236 board by I2C and I3C way.

Signed-off-by: Neil Chen <cheng.chen_1@nxp.com>

hakehuang commented:
@NeilChen93 board test pass, but the twister seems need do below change, as it can not hanlder overlay that smart.

```

sample:
  name: Temperature Sensor
common:
    filter: dt_alias_exists("ambient-temp0")
    harness: console
tests:
  sample.sensor.thermometer:
    tags: sensors
    harness_config:
      fixture: sensor_ambient_temp
      type: multi_line
      regex:
        - "Thermometer Example (.*)"
        - "Temperature device is 0x[0-9|a-z]+, name is [a-z|0-9]+@[a-z|0-9]+"
        - "Temperature is [0-9|.]+°C"
    integration_platforms:
      - nrf52840dk/nrf52840   # mcp9700a
      - frdm_k22f             # tcn75a
      - robokit1              # ntc_thermistor
      - adi_eval_adin1110ebz  # adt7420
    extra_args:
      - platform:mimxrt1180_evk/mimxrt1189/cm33:SHIELD=p3t1755dp_ard_i2c
      - platform:mimxrt1180_evk/mimxrt1189/cm7:SHIELD=p3t1755dp_ard_i2c
      - platform:frdm_mcxn236/mcxn236:SHIELD=p3t1755dp_ard_i2c
  sample.sensor.thermometer_i3c:
    tags: sensors
    harness_config:
      fixture: sensor_ambient_temp_i3c
      type: multi_line
      regex:
        - "Thermometer Example (.*)"
        - "Temperature device is 0x[0-9|a-z]+, name is [a-z|0-9]+@[a-z|0-9]+"
        - "Temperature is [0-9|.]+°C"
    integration_platforms:
      - frdm_mcxn947/mcxn947/cpu0 # p3t1755
    depends_on:
    - i3c
    - sensor
    filter: dt_alias_exists("ambient-temp0")
    extra_args:
      - platform:mimxrt1180_evk/mimxrt1189/cm33:SHIELD=p3t1755dp_ard_i3c
      - platform:mimxrt1180_evk/mimxrt1189/cm7:SHIELD=p3t1755dp_ard_i3c
      - platform:frdm_mcxn236/mcxn236:SHIELD=p3t1755dp_ard_i3c
```
